File: property_managment/public/js/territory.js
frappe.ui.form.on("Territory", {
    refresh: function(frm) {
        console.log("1234567890");

    },
    territory_type: function(frm){
        if(frm.doc.territory_type=="Country" 
        || frm.doc.territory_type=="City" || 
        frm.doc.territory_type=="Town"
        ){
            frm.set_value("is_group", 1);
        }
        else{
            frm.set_value("is_group", 0);

        }
    }

});

File: property_managment/property_company/override/territory.py
import frappe
from erpnext.setup.doctype.territory.territory import Territory


class CustomTerritory(Territory):

    def save(self, *args, **kwargs):
        print("8888888888*100")
        # do_something()
        super().save(*args, **kwargs) # call the base save method
        # do_something_else()

    def validate(self):
        if self.territory_type=="Country" and self.is_group==0:
            frappe.throw("Country can not be is group 1")

        super().validate()




    